d3 Flamegraphs implementation

kornrunner opened the pull request:
Quick & dirty implementation of http://martinspier.io/d3-flame-graph/

markstory commented on src/Xhgui/Profile.php:
```diff
@@ -568,7 +568,82 @@ protected function _callgraphData($parentName, $main, $metric, $threshold, $pare
+        }
+
+        $this->_visited = $this->_nodes = $this->_links = array();
+		$flamegraph = $this->_flamegraphData (self::NO_PARENT, $main, $metric, $threshold);
```
Indentation looks a bit funny here.

markstory commented on src/Xhgui/Profile.php:
```diff
@@ -568,7 +568,82 @@ protected function _callgraphData($parentName, $main, $metric, $threshold, $pare
+
+    protected function _flamegraphData($parentName, $main, $metric, $threshold, $parentIndex = null)
+    {
+ 		$result = [];
```
Indentation is funny.

markstory commented on src/templates/runs/flamegraph.twig:
```diff
@@ -0,0 +1,80 @@
+	  .call(flameGraph);
+});
+
+document.getElementById("form").addEventListener("submit", function(event){
```
XHGui has jQuery available. It might be better to use that so there are no cross-browser DOM issues.

kornrunner commented:
I've tidied code identation, also fixed up search in graph.

To be honest I haven't checked javascript at all initially, basically I've just reused the example from http://martinspier.io/d3-flame-graph/ - I've changed obvious bits to use jQuery instedad in this update.

markstory commented:
Thanks! Merged and tagged in 0.7.0.
